testing/common/data/pool/pool_test.go

```diff
@@ -4,7 +4,6 @@ import (
 	"context"
```

Comment:
please add a test case for a corner case scenario described in one of the comments above